.github/workflows/cmake.yml

```diff
@@ -54,3 +54,28 @@ jobs:
       - name: test
         working-directory: build
         run: ctest --output-on-failure
+
+  build-windows:
+    name: "Build - Windows"
+    runs-on: windows-2022
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install CMake
+        uses: lukka/get-cmake@v3.23.0
+
+      - name: Run vcpkg
+        uses: lukka/run-vcpkg@v11.1
+        with:
+          vcpkgDirectory: ${{ runner.workspace }}/vcpkg
+          vcpkgGitCommitId: 78b61582c9e093fda56a01ebb654be15a0033897 # HEAD on 2023-08-6
+
+      - name: Install jinja
+        run: python -m pip install jinja2
+
+      - name: configure
+        run: cmake -S . -B build -DWITH_JAVA=OFF -DWITH_TESTS=OFF -DCMAKE_TOOLCHAIN_FILE=${{ runner.workspace }}/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_INSTALL_OPTIONS=--clean-after-build -DVCPKG_TARGET_TRIPLET=x64-windows-release -DVCPKG_HOST_TRIPLET=x64-windows-release
```

Comment:
What does the vcpkg toolchain file do?  Should we be building with the various external dependency options set (e.g. USE_SYSTEM_xxx)?

Reply:
It runs `vcpkg install` and add the vcpkg paths to the search dirs. 

> Should we be building with the various external dependency options set (e.g. USE_SYSTEM_xxx)?
We could do that.

Reply:
Should I do that? @PeterJohnson

Reply:
I think we should, as that's more unique to this CI build.

Reply:
I have done that now :)